src/Sarif/Writers/BaseLogger.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Writers
+{
+    public class BaseLogger
+    {
+        private readonly List<FailureLevel> failureLevels;
+        private readonly List<ResultKind> resultKinds;
+
+        public BaseLogger(IEnumerable<FailureLevel> failureLevels,
+            IEnumerable<ResultKind> resultKinds)
+        {
+            this.failureLevels = failureLevels?.ToList() ?? new List<FailureLevel> { FailureLevel.Error, FailureLevel.Warning };
+            this.resultKinds = resultKinds?.ToList() ?? new List<ResultKind> { ResultKind.Fail };
+
+            ValidateParameters();
+        }
+
+        private void ValidateParameters()
+        {
+            if (resultKinds.Any(kind => kind != ResultKind.Fail))
+            {
+                if (failureLevels.Contains(FailureLevel.Error))
+                {
+                    throw new ArgumentException("Invalid kind & level combination");
```

Comment:
>throw new ArgumentException("Invalid kind & level combination"); [](start = 20, length = 64)

This validation is not correct. It's fine to configure a mixture of things. The levels are only *relevant* if kind == Fail.

And so your actual validation here is if Kind does *not* include Fail, then Level *must* be zero.

Reply:
Fixed.

---
In reply to: [559746343](https://github.com/microsoft/sarif-sdk/pull/2241#discussion_r559746343) [](ancestors = 559746343)